### ISSUE TITLE
Inquiry about fixed token postions generated by previous decoding step

### ISSUE THREAD

**Issue by JiHoonLee9898** [ML-GSAI/LLaDA#80]
Hello, thank you for your fascinating works!

I wonder I understood the decoding strategy well.

Regarding your explanation ([https://github.com/ML-GSAI/LLaDA/issues/70#issuecomment-2953911653](url)),

if the block count = 1, decoding steps = 10, target token count = 10, 

the goal of decoding is to generate target 10 tokens and only one token will be generated at each decoding step.

At the first decoding step (t=1, 10 tokens are masked now), llada will predict 10 tokens logits and select the best confidence one. Let's say the selected token is '.' (period), and it is positioned at 10 in 10.

At the second decoding step(t<1, 9 tokens are masked now), llada will predict 9 tokens logits and select the best confidence one. Let's say the selected token is ','(comma),  and it is positioned at 5 in 10.

In this case, 

$M_1$ $M_2$ $M_3$ $M_4$ ,(comma) $M_6$ $M_7$ $M_8$ $M_9$ .(period)

will be the input for the next decoding step (8 tokens are masked now).

Now the output token sequence style is already fixed to **a sentence consisting of two clauses, where each clause consists exact 4 tokens**.

If tokens that limit the length of a sentence or a clause (comma or period tokens) appear early like this, 

the possible token combinations to represent each clause of sentence would become very limited. 

Is my understanding correct?

**Comment by NieShenRuc** [comment]
Thanks for your interest! I think your description of the generation process is correct.

However, I may not have fully understood what you meant by “the possible token combinations to represent each clause of the sentence would become very limited.”

You can refer to the GIF in Question 5 of the FAQ section in the README.md. Even in the early stages of sampling (for example, when the model samples a comma or period first, as you mentioned), the model still maintains a plan for what content should be sampled in the remaining positions.
